flight/Modules/GPS/GPS.c

```diff
@@ -229,15 +207,44 @@ static void gpsTask(void *parameters)
 				ubx_cfg_send_configuration(gpsPort, gps_rx_buffer);
 			}
 		}
-			break;
+		break;
 #endif
 	}
 #endif /* PIOS_GPS_MINIMAL */
+}
+
+/**
+ * Main gps task. It does not return.
+ */
+
+static void gpsTask(void *parameters)
+{
+	GPSPositionData gpsposition;
+
+	uint32_t timeOfLastUpdateMs = 0;
+	uint32_t timeOfConfigAttemptMs = 0;
+
+	uint8_t	gpsProtocol;
+
+#ifdef PIOS_GPS_PROVIDES_AIRSPEED
+	gps_airspeed_initialize();
+#endif
 
 	GPSPositionGet(&gpsposition);
+
 	// Loop forever
-	while (1)
-	{
+	while (1) {
+		uint32_t xDelay = GPS_COM_TIMEOUT_MS;
```

Comment:
Not related to your changes, but I'd love to see this changed to a descriptive name like ```characterTimeoutMs```